### PR TITLE
Adds sound effects to surgeries

### PR DESCRIPTION
**Pull request by maxspells**
## About The Pull Request

Adds sound effects to tools, surgeries, removing organs, etc.

This is not finished yet, and I PR'd it early by mistake but I'll leave it open as a draft for now since the code is working at the moment, and I can just expand it and make the changes I want here I guess.

Here's a video example of an Organ Manipulation surgery with the current placeholder sound effects:
[Organ Manipulation+Tendwounds w/sound](https://www.youtube.com/watch?v=k3dFFFAMLD0)
## Why It's Good For The Game

Staring at progress bars is boring. This will hopefully make performing surgeries more engaging for MDs and feel less monotonous.

## Changelog

:cl:
expansion: Surgery steps play sound effects appropriate to the step/tool being used
expansion: Added flavortext for clamping bleeders
expansion: Added success code/flavortext for retracting skin
expansion: Added success code/flavortext for some mechanical surgery steps
expansion: Implements sound effects for mechanical surgery
soundadd: Surgery tool sound effects
soundadd: Organ/surgery sound effects
/:cl:

### CODE REVIEW

**Comment by maxspells** [comment]
oops

**Comment by Fikou** [comment]
yoo surgery dropped

**Comment by AdipemDragon** [comment]
loved seeing the pr closing speedrun any% live

**Comment by maxspells** [comment]
yeah I'm new at this, was trying to open a draft to my fork so I could send to my friend to look over and accidentally PR'd it here lmao

**Comment by Iamgoofball** [comment]
@maxspells Would you be willing to re-open this? Adding sound effects to Surgery is good.

**Comment by maxspells** [comment]
> @maxspells Would you be willing to re-open this? Adding sound effects to Surgery is good.

Yes but, there are a lot of things I'll have to work on first. This wasn't supposed to be PR'd yet, I just got the code and sounds to work and wanted to send to my friends.

**Comment by Iamgoofball** [comment]
:+1:

**Comment by exymian** [comment]
speedmerge

**Comment by ArcaneMusic** [comment]
Unironically think this is neat even if you were planning on packing more into this PR.

**Comment by maxspells** [comment]
Issues I'm looking at would be if alien tools+advanced surgery would be an overload of sound, and of course 3-4 MDs all doing surgery in the same room might be super gross to listen to for an entire shift. 

I have a couple ideas to fix those, and I also want to add more sound effects for like, laser scalpels and stuff.

**Comment by SpaceDragon00** [comment]
I find it a bit odd how some of the sfx start when the step is completed, and some start when the step begins. I think it'd be a lot better if most of the sounds started with the step

**Comment by Mothblocks** [comment]
What are the sources for these sounds?

(Also I see the circular saw sound getting *really* annoying, but I'm not exactly sure what else you'd pick, so clamping the frequencies will need to be done at some point)

**Comment by maxspells** [comment]
> What are the sources for these sounds?
> 
The original samples came from a variety of places, but the short answer is I made all of them myself. None of these are stock samples and you won't be able to find them anywhere, (except maybe the saw one)

If needed I can post the FL studio files I made them in, but I'm not real happy with some of them and I plan on doing more and replacing some when I get home from work tonight

**Comment by Mothblocks** [comment]
:+1: Thanks! Licensing was my concern.

**Comment by maxspells** [comment]
> I find it a bit odd how some of the sfx start when the step is completed, and some start when the step begins. I think it'd be a lot better if most of the sounds started with the step

So this has been something that has hopefully been fixed. All sound effects should now have separate sfx that play during/at the beginning of a step and another one when the step is successful.

**Comment by Mothblocks** [comment]
Undraft if ready

**Comment by maxspells** [comment]
I'm just going to open this. I wanted to get this working with ghetto surgery but I just don't have the time to expand this as much as I want to right now, so I'm going to put it up and see what you guys think in the current state and maybe expand it with another PR later

File: code/modules/surgery/organic_steps.dm [hunk 1]
```diff
@@ -79,9 +86,18 @@
...
-		span_notice("[user] begins to retract the skin in [target]'s [parse_zone(target_zone)]."))
+		span_notice("[user] begins to retract the skin in [target]'s [parse_zone(target_zone)]."),
+		playsound(get_turf(target), 'sound/surgery/retractor1.ogg', 75, TRUE, falloff_exponent = 12, falloff_distance = 1))
 	display_pain(target, "You feel a severe stinging pain spreading across your [parse_zone(target_zone)] as the skin is pulled back!")
 
+/datum/surgery_step/retract_skin/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results)
+	display_results(user, target, span_notice("You retract the skin from [target]'s [parse_zone(target_zone)]."),
+		span_notice("[user] retracts the skin from [target]'s [parse_zone(target_zone)]."),
+		span_notice("[user] retracts the skin from [target]'s [parse_zone(target_zone)]."),
+		playsound(get_turf(target), 'sound/surgery/retractor2.ogg', 75, TRUE, falloff_exponent = 12, falloff_distance = 1))
```
Comment (Mothblocks) [inline code/modules/surgery/organic_steps.dm:97]: Why are these all treated as arguments to display_results? This makes no sense, make it its own line.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by LemonInTheDark** [comment]
Why are the cautery and saw 1 files so much larger then the others?

**Comment by maxspells** [comment]
> Why are the cautery and saw 1 files so much larger then the others?

If I had to guess, it's because they are longer and are meant to play during the majority of the progress bar instead of a quick sound effect snip at the beginning

File: code/modules/surgery/organic_steps.dm (hunk 1 above)
Comment (maxspells) [reply code/modules/surgery/organic_steps.dm:97]: Okay I'll move them over!

Comment (nmajask) [reply code/modules/surgery/organic_steps.dm:97]: You could also make the sound files a variable of surgery_step and add the playsound() to initiate() to cut down on copy pasted code.

**Comment by maxspells** [comment]
Closing this. I'm busy in life and don't have the time to do this. Maybe I'll PR this later when things slow down. Thanks for the criticisms! Sorry I couldn't make the changes needed :)

**Comment by Iamgoofball** [comment]
its ok u did good work

**Comment by nmajask** [comment]
@maxspells Do you mind if I open another PR reviving this, crediting you for the sounds of course

**Comment by maxspells** [comment]
Sure go for it!

**Pull request closed without merging**